File: services/reservation/src/main/java/com/example/iter/reservation/dto/response/RentalCancelResponse.java
package com.example.iter.reservation.dto.response;

import com.example.iter.payment.api.PaymentStatus;
import com.example.iter.reservation.domain.entity.Rental;
import com.example.iter.reservation.api.RentalStatus;

public record RentalCancelResponse(
        Long rentalId,
        RentalStatus status,
        PaymentStatus paymentStatus
) {
    public static RentalCancelResponse of(Rental rental, PaymentStatus paymentStatus) {
        return new RentalCancelResponse(rental.getId(), rental.getStatus(), paymentStatus);
    }
}

File: services/reservation/src/main/kotlin/com/example/iter/reservation/dto/response/RentalCancelResponse.kt
package com.example.iter.reservation.dto.response

import com.example.iter.payment.api.PaymentStatus
import com.example.iter.reservation.api.RentalStatus
import com.example.iter.reservation.domain.entity.Rental

data class RentalCancelResponse(
    val rentalId: Long?,
    val status: RentalStatus?,
    val paymentStatus: PaymentStatus?,
) {
    fun rentalId(): Long? = rentalId
    fun status(): RentalStatus? = status
    fun paymentStatus(): PaymentStatus? = paymentStatus

    companion object {
        @JvmStatic
        fun of(rental: Rental, paymentStatus: PaymentStatus?): RentalCancelResponse =
            RentalCancelResponse(rental.id, rental.status, paymentStatus)
    }
}
